File: pset6/credit.py
from cs50 import get_int
from math import floor
no = get_int("enter your credit card no. : ")

sum1,sum2 = 0,0
if floor(no/10**17)==0:
    for i in range(15,0,-2):
        sum11 = floor(no/(10**i))
        sum1 = sum1 + (2*(sum11//10))

    for j in range(16,-1,-2):
        sum22 = floor(no/(10**i))
        sum2 = sum2 + (sum22//10)
else:
    for i in range(14,1,-2):
        sum11 = floor(no/(10**i))
        sum1 = sum1 + (2*(sum11//10))

    for j in range(15,0,-2):
        sum22 = floor(no/(10**i))
        sum2 = sum2 + (sum22//10)

total = sum1 + sum2

if((total//10)==0):
    print('valid credit card')

else:
    print('invalid')

File: pset1/credit.c
#include<stdio.h>
#include<cs50.h>
#include<math.h>

int main(void)
{
    long int n;
    int sum=0;
    int sum1=0;
    int sum2=0;
    long int R=0,y=0;
    while(1)
    {
        n= get_long_long("enter your card number :");
        if(n>0)
        {
            break;
        }
        else
        {
            continue;
        }
    }
    for(int i=1;i<=13;i+=2)
    {
        R=n/pow(10,i);
        sum1+=(2*(R%10));
    }
    for(int J=0;J<=15;J+=2)
    {
        y=n/pow(10,J);
        sum2+=(y%10);
    }
    sum=sum1+sum2;
    if (sum%10==0)
    {
        printf("VISA\n");
    }
    else printf("INVALID\n");
}